quickstart/101-vm-auto-shutdown/variables.tf

```diff
@@ -0,0 +1,93 @@
+variable "resource_group_location" {
+  description = "The location where the resource group should be created."
+  type        = string
+  default     = "East US"
+}
+
+variable "prefix" {
+  description = "A prefix for naming resources."
+  type        = string
+  default     = "demo"
+}
+
+variable "vnet_address_space" {
+  description = "Address space for the virtual network."
+  type        = list(string)
+  default     = ["10.0.0.0/16"]
+}
+
+variable "subnet_address_prefixes" {
+  description = "Address prefixes for the subnet."
+  type        = list(string)
+  default     = ["10.0.1.0/24"]
+}
+
+variable "public_ip_allocation_method" {
+  description = "Allocation method for the public IP."
+  type        = string
+  default     = "Dynamic"
+}
+
+variable "vm_size" {
+  description = "Size of the virtual machine."
+  type        = string
+  default     = "Standard_DS1_v2"
+}
+
+variable "vm_image" {
+  description = "Source image reference for the virtual machine."
+  type = object({
+    publisher = string
+    offer     = string
+    sku       = string
+    version   = string
+  })
+  default = {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2022-datacenter-azure-edition"
+    version   = "latest"
+  }
+}
+
+variable "storage_account_tier" {
+  description = "Performance tier of the storage account."
+  type        = string
+  default     = "Standard"
+}
+
+variable "storage_account_replication_type" {
+  description = "Replication type for the storage account."
+  type        = string
+  default     = "LRS"
+}
+
+variable "automation_account_sku_name" {
```

Comment:
Please include a validation block for all restricted values. (Example: https://github.com/Azure/terraform/blob/master/quickstart/101-front-door-standard-premium/variables.tf)

Reply:
Thanks for the catch. Updated in commit: https://github.com/Azure/terraform/pull/255/commits/f46699d1f220ccf21bb974cb7faecb545fd08206